[Bug]: `tacacs/test_authorization.py::test_authorization_tacacs_only_then_server_down_after_login` fails with `Failed: Not found aaa config update log: sudo config aaa authorization tacacs+`

### Issue Description

```
    def change_and_wait_aaa_config_update(duthost, command, last_timestamp=None, timeout=10):
        if not last_timestamp:
            last_timestamp = get_auditd_config_reload_timestamp(duthost)

        duthost.shell(command)

        # After AAA config update, hostcfgd will modify config file and notify auditd reload config
        # Wait auditd reload config finish
        def log_exist(duthost):
            latest_timestamp = get_auditd_config_reload_timestamp(duthost)
            return latest_timestamp != last_timestamp

        exist = wait_until(timeout, 1, 0, log_exist, duthost)
>       pytest_assert(exist, "Not found aaa config update log: {}".format(command))
E       Failed: Not found aaa config update log: sudo config aaa authorization tacacs+
```

The test grabs the last instance of the log returned by `sudo journalctl -u auditd --boot --no-pager | grep 'audisp-tacplus re-initializing configuration'`

Then enables `sudo config aaa authorization tacacs+`

And confirms there is a newer instance of the `audisp-tacplus re-initializing configuration` log.

However, the issue is it checks for this newer instance based on the timestamp, but the timestamp is at a per-second granularity.
So if the new log is within the same second the test will fail.

E.G. from our output:
```
29/01/2025 00:26:55 base._run                                L0071 DEBUG  | /data/tests/common/devices/multi_asic.py::_run_on_asics#136: [cmp227-4] AnsibleModule::shell, args=["sudo journalctl -u auditd --boot --no-pager | grep 'audisp-tacplus re-initializing configuration'"], kwargs={}
29/01/2025 00:26:56 base._run                                L0108 DEBUG  | /data/tests/common/devices/multi_asic.py::_run_on_asics#136: [cmp227-4] AnsibleModule::shell Result => {"changed": true, "stdout":
   "Jan 28 10:02:12 cmp227-4 audisp-tacplus[773]: audisp-tacplus re-initializing configuration\n
   ...
   "Jan 29 00:26:54 cmp227-4 audisp-tacplus[773]: audisp-tacplus re-initializing configuration",
   "Jan 29 00:26:55 cmp227-4 audisp-tacplus[773]: audisp-tacplus re-initializing configuration",
   "Jan 29 00:26:56 cmp227-4 audisp-tacplus[773]: audisp-tacplus re-initializing configuration"]

29/01/2025 00:26:56 base._run                                L0071 DEBUG  | /data/tests/common/devices/multi_asic.py::_run_on_asics#136: [cmp227-4] AnsibleModule::shell, args=["sudo config aaa authorization tacacs+"], kwargs={}

29/01/2025 00:26:56 base._run                                L0071 DEBUG  | /data/tests/common/devices/multi_asic.py::_run_on_asics#136: [cmp227-4] AnsibleModule::shell, args=["sudo journalctl -u auditd --boot --no-pager | grep 'audisp-tacplus re-initializing configuration'"], kwargs={}
29/01/2025 00:26:57 base._run                                L0108 DEBUG  | /data/tests/common/devices/multi_asic.py::_run_on_asics#136: [cmp227-4] AnsibleModule::shell Result => {"changed": true, "stdout":
   "Jan 28 10:02:12 cmp227-4 audisp-tacplus[773]: audisp-tacplus re-initializing configuration\n
   ...
   "Jan 29 00:26:54 cmp227-4 audisp-tacplus[773]: audisp-tacplus re-initializing configuration",
   "Jan 29 00:26:55 cmp227-4 audisp-tacplus[773]: audisp-tacplus re-initializing configuration",
   "Jan 29 00:26:56 cmp227-4 audisp-tacplus[773]: audisp-tacplus re-initializing configuration",
   "Jan 29 00:26:56 cmp227-4 audisp-tacplus[773]: audisp-tacplus re-initializing configuration"]
```

Note that there is a new entry but because it has the same timestamp `Jan 29 00:26:56` as the previous entry the test will hit a string match and cause the test to fail:
```
def log_exist(duthost):
    latest_timestamp = get_auditd_config_reload_timestamp(duthost)
    return latest_timestamp != last_timestamp
```

### Results you see

Test failing when back to back logs are within the same second.

### Results you expected to see

Test should pass if it sees a new log, we shouldn't rely on the timestamp.

### Is it platform specific

generic

### Relevant log output

```shell

```

### Output of `show version`

```shell

```

### Attach files (if any)

_No response_

--- a/tests/tacacs/test_accounting.py
+++ b/tests/tacacs/test_accounting.py
@@ -4,7 +4,7 @@ import pytest
 from tests.common.devices.ptf import PTFHost
 from tests.common.helpers.tacacs.tacacs_helper import stop_tacacs_server, start_tacacs_server, \
     per_command_accounting_skip_versions, remove_all_tacacs_server
-from .utils import check_server_received, change_and_wait_aaa_config_update, get_auditd_config_reload_timestamp, \
+from .utils import check_server_received, change_and_wait_aaa_config_update, get_auditd_config_reload_line_count, \
     ensure_tacacs_server_running_after_ut, ssh_connect_remote_retry, ssh_run_command    # noqa: F401
 from tests.common.errors import RunAnsibleModuleFail
 from tests.common.helpers.assertions import pytest_assert
@@ -253,7 +253,7 @@ def test_accounting_tacacs_only_some_tacacs_server_down(
     # when tacacs config change multiple time in short time
     # auditd service may been request reload during reloading
     # when this happen, auditd will ignore request and only reload once
-    last_timestamp = get_auditd_config_reload_timestamp(duthost)
+    last_line_count = get_auditd_config_reload_line_count(duthost)
 
     duthost.shell("sudo config tacacs timeout 1")
     remove_all_tacacs_server(duthost)
@@ -261,7 +261,7 @@ def test_accounting_tacacs_only_some_tacacs_server_down(
     duthost.shell("sudo config tacacs add %s --port 59" % tacacs_server_ip)
     change_and_wait_aaa_config_update(duthost,
                                       "sudo config aaa accounting tacacs+",
-                                      last_timestamp)
+                                      last_line_count)
 
     cleanup_tacacs_log(ptfhost, rw_user_client)
 

--- a/tests/tacacs/utils.py
+++ b/tests/tacacs/utils.py
@@ -65,27 +65,24 @@ def check_server_received(ptfhost, data, timeout=30):
     pytest_assert(exist, "Not found data: {} in tacplus server log".format(data))
 
 
-def get_auditd_config_reload_timestamp(duthost):
+def get_auditd_config_reload_line_count(duthost):
     res = duthost.shell("sudo journalctl -u auditd --boot --no-pager | grep 'audisp-tacplus re-initializing configuration'") # noqa E501
     logger.info("aaa config file timestamp {}".format(res["stdout_lines"]))
 
-    if len(res["stdout_lines"]) == 0:
-        return ""
-
-    return res["stdout_lines"][-1]
+    return len(res["stdout_lines"])
 
 
-def change_and_wait_aaa_config_update(duthost, command, last_timestamp=None, timeout=10):
-    if not last_timestamp:
-        last_timestamp = get_auditd_config_reload_timestamp(duthost)
+def change_and_wait_aaa_config_update(duthost, command, last_line_count=None, timeout=10):
+    if not last_line_count:
+        last_line_count = get_auditd_config_reload_line_count(duthost)
 
     duthost.shell(command)
 
     # After AAA config update, hostcfgd will modify config file and notify auditd reload config
     # Wait auditd reload config finish
     def log_exist(duthost):
-        latest_timestamp = get_auditd_config_reload_timestamp(duthost)
-        return latest_timestamp != last_timestamp
+        latest_line_count = get_auditd_config_reload_line_count(duthost)
+        return latest_line_count > last_line_count
 
     exist = wait_until(timeout, 1, 0, log_exist, duthost)
     pytest_assert(exist, "Not found aaa config update log: {}".format(command))

